apps/emqx_bridge/test/emqx_bridge_api_SUITE.erl

```diff
@@ -54,38 +57,124 @@
     <<"method">> => <<"post">>,
     <<"body">> => <<"${payload}">>,
     <<"headers">> => #{
-        <<"content-type">> => <<"application/json">>
+        % NOTE
+        % The Pascal-Case is important here.
+        % The reason is kinda ridiculous: `emqx_bridge_resource:create_dry_run/2` converts
+        % bridge config keys into atoms, and the atom 'Content-Type' exists in the ERTS
+        % when this happens (while the 'content-type' does not).
+        <<"Content-Type">> => <<"application/json">>
     }
 }).
 -define(HTTP_BRIDGE(URL), ?HTTP_BRIDGE(URL, ?BRIDGE_NAME)).
 
 all() ->
-    emqx_common_test_helpers:all(?MODULE).
+    [
+        {group, single},
+        {group, cluster}
+    ].
 
 groups() ->
-    [].
+    SingleOnlyTests = [
+        t_broken_bpapi_vsn,
+        t_old_bpapi_vsn,
+        t_bridges_probe
+    ],
+    [
+        {single, [], emqx_common_test_helpers:all(?MODULE)},
+        {cluster, [], emqx_common_test_helpers:all(?MODULE) -- SingleOnlyTests}
+    ].
 
 suite() ->
     [{timetrap, {seconds, 60}}].
 
 init_per_suite(Config) ->
-    _ = application:load(emqx_conf),
-    %% some testcases (may from other app) already get emqx_connector started
-    _ = application:stop(emqx_resource),
-    _ = application:stop(emqx_connector),
-    ok = emqx_mgmt_api_test_util:init_suite(
-        [emqx_rule_engine, emqx_bridge, emqx_authn]
-    ),
-    ok = emqx_common_test_helpers:load_config(
-        emqx_rule_engine_schema,
-        <<"rule_engine {rules {}}">>
-    ),
-    ok = emqx_common_test_helpers:load_config(emqx_bridge_schema, ?CONF_DEFAULT),
     Config.
 
 end_per_suite(_Config) ->
-    emqx_mgmt_api_test_util:end_suite([emqx_rule_engine, emqx_bridge, emqx_authn]),
-    mria:clear_table(emqx_authn_mnesia),
+    ok.
+
+init_per_group(cluster, Config) ->
+    ok = snabbkaffe:fix_ct_logging(),
+    Cluster = mk_cluster_specs(Config),
+    ct:pal("Starting ~p", [Cluster]),
+    Nodes = [
+        emqx_common_test_helpers:start_slave(Name, Opts)
+     || {Name, Opts} <- Cluster
+    ],
+    [NodePrimary | NodesRest] = Nodes,
+    ok = erpc:call(NodePrimary, fun() -> init_node(primary) end),
+    _ = [ok = erpc:call(Node, fun() -> init_node(regular) end) || Node <- NodesRest],
+    [{group, cluster}, {cluster_nodes, Nodes}, {api_node, NodePrimary} | Config];
+init_per_group(_, Config) ->
+    ok = emqx_mgmt_api_test_util:init_suite(?SUITE_APPS),
+    ok = load_suite_config(emqx_rule_engine),
+    ok = load_suite_config(emqx_bridge),
+    [{group, single}, {api_node, node()} | Config].
+
+mk_cluster_specs(Config) ->
+    Specs = [
+        {core, emqx_bridge_api_SUITE1, #{}},
+        {core, emqx_bridge_api_SUITE2, #{}}
+    ],
+    CommonOpts = #{
+        env => [{emqx, boot_modules, [broker]}],
+        apps => [],
+        % NOTE
```

Comment:
Great catch!

I worked around this (similar?) problem with this: https://github.com/keynslug/emqx/blob/89822c7a46c1f33aab9a268e2affd1b5db38e5ae/lib-ee/emqx_ee_bridge/test/emqx_bridge_impl_kafka_consumer_SUITE.erl#L1063

Maybe a bit hackish, but seemed to work :see_no_evil:

Reply:
Hah, nice one, seen it but had no idea what it does. 😅